src/ruby/ext/grpc/rb_server.c

```diff
@@ -326,7 +327,14 @@ static VALUE grpc_rb_server_add_http2_port(VALUE self, VALUE port,
                StringValueCStr(port));
     }
   } else {
-    creds = grpc_rb_get_wrapped_server_credentials(rb_creds);
+    if (grpc_rb_is_server_credentials(rb_creds)) {
+      creds = grpc_rb_get_wrapped_server_credentials(rb_creds);
+    } else if (grpc_rb_is_xds_server_credentials(rb_creds)) {
+      creds = grpc_rb_get_wrapped_xds_server_credentials(rb_creds);
+    } else {
+      rb_raise(rb_eTypeError,
+               "bad creds, want ServerCredentials or XdsServerCredentials");
```

Comment:
nit: can we specify this error a bit more, to something like:

`"failed to create server because credentials parameter has an invalid type, want ServerCredentials or XdsServerCredentials"`

Reply:
done